x-pack/plugins/security_solution/server/lib/detection_engine/routes/index/read_signal_index_route.ts

```diff
@@ -0,0 +1,96 @@
+/*
```

Comment:
This api is similar to `x-pack/plugins/security_solution/server/lib/detection_engine/routes/index/read_index_route.ts`, only thing different is this uses `getIndexExists` to check indexExists.

I expected `read_index_route` could check if alerts data exists, but it's not working as what I expected, so create this new route to do the work, but any idea is welcome.

Reply:
This route seems like it shouldn't have much duplicated logic with `read_index_route` if it only needs to check if the index exists. I'd suggest limiting it to something like
```
const indexName = ruleDataService.getResourceName(`security.alerts-${spaceId}`);
const indexExists = await getIndexExists(core.elasticsearch.client.asInternalUser, indexName);
return response.ok({
  body: {
    indexExists,
  },
});
```
as the core logic, and remove the whole `if (indexExists) {` block. This keeps the mapping update logic contained in `read_index_route`.

Reply:
I would also suggest renaming this route to `read_alerts_index_exists_route`.

Reply:
Thank you @marshallmain , I'll update it accordingly.